### PR TITLE
Add internal/otelarrow package for shared code between otelarrow exporter, receiver

### PR DESCRIPTION
**Pull request by jmacd**
**Description:** The otelarrowexporter and otelarrowreceiver components depend on several libraries in the open-telemetry/otel-arrow repository, which would complicate release management in both repositories. This change eliminates the dependency on `otel-arrow/collector` by moving each of those packages into a subdirectory of `internal/otelarrow`.

**Link to tracking Issue:** #33567 

**Testing:** This brings new integration tests into this repository for the two otelarrow components.

**Documentation:** Each of the three non-test packages exposed here has a brief README.md explaining why it exists. Each of these could be useful to non-Arrow components if helpful.

### CODE REVIEW

**Review by moh-osman3 — APPROVED** [review]

**Review by lquerel — APPROVED** [review]
LGTM

**Comment by jmacd** [comment]
Reviewers -- this is not stale. This PR is blocking future development and minor improvements of the otel-arrow components. This is also intended to avoid dependency skew between these packages (which depend on unstable collector APIs but reside in an external repository) before they begin to hold up releases in this repository.

File: internal/otelarrow/netstats/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Network Statistics package
+
+## Overview
+
+This package provides general support for couning the number of bytes
```
Comment (mwear) [inline internal/otelarrow/netstats/README.md:5]:
```suggestion
This package provides general support for counting the number of bytes
```

**Review by mwear — APPROVED** [review]
LGTM. One non-blocking question and a couple of suggestions.

File: internal/otelarrow/compression/zstd/mru.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	now := time.Now()
+
+	mru.freelist = append(mru.freelist, item)
+	mru.putTimes = append(mru.putTimes, now)
+
+	// Evict any objects that haven't been touched recently.
+	for len(mru.putTimes) > 0 && now.Sub(mru.putTimes[0]) >= TTL {
+		// Shift values by one index in the slice, to preserve capacity.
+		l := len(mru.freelist)
+		copy(mru.freelist[0:l-1], mru.freelist[1:])
```
[the remark this answers is not in the document]
Comment (jmacd) [reply internal/otelarrow/compression/zstd/mru.go:85]: Container/list would allocate more objects-- this is a slice-based to reduce allocations. This little bit of code was extracted (with permission) from a heavily-used internal piece of code, so it arrives with little justification. I'm willing to consider changes.

[the remark this answers is not in the document]
Comment (mwear) [reply internal/otelarrow/compression/zstd/mru.go:85]: It's a tradeoff and it appears as though it wouldn't be worth it. I'm fine with it as is.

**Review by codeboten — APPROVED** [review]
Opened a follow up issue to move the usage of otel metrics to mdatagen https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34135

**Pull request merged**